File: app/src/java/test/java/com/vktiwari/pagingretrofitrxjava/NetworkStateTest.java
package com.vktiwari.pagingretrofitrxjava;

import com.vktiwari.pagingretrofitrxjava.network.NetworkState;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class NetworkStateTest {
    @Test
    public void testErrorNetworkState(){
        NetworkState state = NetworkState.error("test");
        assertEquals(state.getStatus(), NetworkState.Status.FAILED);
        assertEquals("test",state.getMessage());
    }
}

File: app/src/testKotlin/java/com/vktiwari/pagingretrofitrxjava/NetworkStateTest.kt
package com.vktiwari.pagingretrofitrxjava

import com.vktiwari.pagingretrofitrxjava.network.NetworkState
import org.junit.Assert.assertEquals
import org.junit.Test

class NetworkStateTest {
    @Test
    fun testErrorNetworkState() {
        val state = NetworkState.error("test")
        assertEquals(state.getStatus(), NetworkState.Status.FAILED)
        assertEquals("test", state.getMessage())
    }
}